src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -123,8 +123,9 @@ internal class SqlServerFhirDataStore : IFhirDataStore, IProvideCapability
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
+                // ** We must use CreateNonRetrySqlCommand here because the retry will not reset the Stream containing the RawResource, resulting in a failure to save the data
                 using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, true))
-                using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateRetrySqlCommand())
+                using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateNonRetrySqlCommand())
```

Comment:
Is this the only command that need retry disabled? We use this in a lot of places.

Reply:
Taking a quick look I think we should also disallow retries in the SqlServerTaskManager line 39, SqlServerContextUpdater line 46, SqlServerTaskConsumer lines 52, 172, and 229, SqlServerFhirOperationsDataStore lines 57, 130, 200, 249, SqlServerFhirModel lines 227 and 373, SqlServerFhirDataStore lines 498 and 615, and SqlServerSearchParameterStatusDataStore line 205.

Reply:
All the commands above are creating or updating resources, so they could have the same stream issue.

Reply:
Its only a problem if we use Stream

Reply:
I think this is ok to merge. Of all the other instances I've checked, RawResource is the only instance I can see that uses VarBinary / Stream. VarChar / String is fine